### DISCUSSION TITLE
`simple-markdown` is not listed as a dependency/peer dependency

### DISCUSSION THREAD

**Discussion by joelpierre** [GetStream/stream-chat-react-native#1803] [General]
**Describe the bug**
Peer dependency for `simple-markdown` not specified in any package.json.

**Dev environment info (please complete/provide the following information):**

- Package version - v5.2.0
- react-native
- Device/Emulator (android/ios) and OS version - BOTH

**To Reproduce**
Run a dev build or standard build

**Expected behavior**
You should list it as a peer dependency so we know to install it and what version

**Screenshots**

![Screenshot 2022-11-08 at 10 49 40](https://user-images.githubusercontent.com/8969851/200551656-67831aa5-3ec2-4236-a162-37ea0df0b3ab.png)

**Comment by madsroskar** [comment]
`simple-markdown` is a dependency of `react-native-markdown-package`, which in turn is a dependency of `stream-chat-react-native-core`, and should be available through your package manager when that is resolved. Are you able to make a minimal reproduction of this issue? When I set up a simple app using the SDK, I'm not able to reproduce an issue of `simple-markdown` not being available.

**Comment by joelpierre** [comment]
Yeah I noticed that too. So was surprised it wasn't "auto" resolved. We are however using a mono repo with yarn workspaces with an installConfig.hoistingLimit set to `dependencies`. Theoretically it should install all of your modules packages regardless of this setting however it is missing.

Our fix for now is just to add that as a dependency for us. However with that being said, if you use the package directly in your core package I think you should reference it in your package.json?

I can't get to minimal reproduction for now 😅

**Comment by vishalnarkhede** [comment]
@joelpierre Since it's not a blocker for you are the moment, I am going to move it to a discussion for anyone to add more suggestions to this problem.

**Comment by vishalnarkhede** [comment]
We could list these packages (which dependencies of dependencies) as dependencies of our core project. But I feel like it will make it super hard to manage and maintain dependency versions. 

Open for better suggestions.

**Comment by joelpierre** [comment]
Thats fine @vishalnarkhede , probably falling more into preference territory for sure, but in general when I build packages like this I like to be explicit with anything I use and thus import into components/utils etc. Often I have got into situations in which I rely on something another package brings along with it and then a minor/patch version later they remove their dependency on it and I am 😢 because I depended on them depending on it 😆 .

Version management complexities is definitely the down side, however I am more in control of that than an external packages dependencies between versions.
